block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -182,11 +172,8 @@ impl BlockWitness {
         block_inputs: &BlockInputs,
         batches: &[TransactionBatch],
     ) -> Result<(), BuildBlockError> {
-        let produced_nullifiers_from_store: BTreeSet<Nullifier> = block_inputs
-            .nullifiers
-            .iter()
-            .map(|nullifier_record| nullifier_record.nullifier)
-            .collect();
+        let produced_nullifiers_from_store: BTreeSet<Nullifier> =
+            block_inputs.nullifiers.iter().map(|(&nullifier, _)| nullifier).collect();
```

Comment:
Maybe use the `keys` API?

```suggestion
            block_inputs.nullifiers.keys().collect();
```